File: lab12/prelab12/linkedlist.m
/*
John Zheng
jz5pt
4/28/19
linkedlist.m
*/
#import <stdio.h>
#import <Foundation/NSObject.h>

//ListNode interface set up
@interface ListNode : NSObject {
	@private
		int val;
		ListNode* next;
}
- (id) val: (int) val_value;
- (int) val;
- (id) next: (ListNode*) nextPtr;
- (ListNode*) next;
@end

//implementing the ListNode methods
@implementation ListNode
//val setter
- (id) val: (int) val_value {
	val = val_value;
	return self;
}
//val getter
- (int) val {
	return val;
}
//next pointer setter
- (id) next: (ListNode*) nextPtr {
	next = nextPtr;
	return self;
}
//next pointer getter
- (ListNode*) next {
	return next;
}
@end

//List interface set up
@interface ListClass : NSObject {
	@private 
		ListNode* head;
}
- (id) head: (ListNode*) headPtr;
- (ListNode*) head;
@end

//implementing List 
@implementation ListClass
//head setter
- (id) head: (ListNode*) headPtr {
	head = headPtr;
	return self;
}
//head getter
- (ListNode*) head {
	return head;
}
@end

int main (void) {
	ListClass *list = [ListClass new];
	[list head: nil];
	int n;
	printf("Enter how many values to input:	");
	scanf("%d", &n);
	int num;
	for (int i = 0; i < n; i++) {
		printf("Enter value:	");
		scanf("%d", &num);
		ListNode *node = [ListNode new];
		[node val: num];
		[node next: [list head]];
		[list head: node];
	}
	ListNode* itr = [list head];
	for (int i = 0; i < n; i++) {
		printf("%d\n", [itr val]);
		itr = [itr next];
	}
	
	[list release];
	[itr release];
    return 0;
}

File: lab9/postlab9/linkedlist.c
/*
John Zheng
jz5pt
linkedlist.c
4/12/19
*/

#include <stdio.h>

struct list_item {
	struct list_item *next;
	int val;
};

struct list {
	struct list_item *head;
};

int main() {
	struct list *theList;
	struct list_item *node;
	int n;
	printf("Enter how many values to input: ");
	scanf("%d", &n);	
	theList = (struct list*) malloc(sizeof(struct list));
	theList->head = NULL;
	int num;
	//adds node to list
	for (int i = 0; i < n; i++) {
		printf("Enter value: ");
		scanf("%d", &num);
		node = (struct list_item*) malloc(sizeof(struct list_item));
		node->val = num;
		node->next = theList->head;
		theList->head = node;
	}
	
	//prints list
	struct list_item *itr;
	itr = theList->head;
	for (int i = 0; i < n; i++) {
		printf("%d\n",itr->val);
		itr = itr->next;
	}

	free(theList);
	free(node);

	return 0;
}
